app/src/utils/clients/ReportDbClient.ts

```diff
@@ -0,0 +1,187 @@
+import { IDBReport, IReportClient, IReport } from '@/models';
+import { AttributeValue, DynamoDB } from '@aws-sdk/client-dynamodb';
+import { getDynamoDbClient } from '../api';
+import { getUuid, isUndefined } from '../common';
+import { getStage, getTenant } from '../environment';
+
+const BASE_TABLE_NAME = 'ReportsTable';
+const CATEGORY_INDEX_NAME = 'CategoryIndex';
+
+/** Client to interact with report DynamoDB */
+export class ReportDbClient implements IReportClient {
+    ddbClient: DynamoDB;
+
+    constructor() {
+        this.ddbClient = getDynamoDbClient();
+    }
+
+    /** Create report */
+    async createReport(report: Omit<IReport, 'reportId'>) {
+        const reportId = getUuid();
+
+        await this.ddbClient.putItem({
+            TableName: getTableName(),
+            Item: marshalReport({
+                ...report,
+                reportId,
+            }),
+        });
+
+        return this.getReport(reportId);
+    }
+
+    /** Get report */
+    async getReport(reportId: string) {
+        const key: Omit<
+            IDBReport,
+            | 'Name'
+            | 'EmailAddress'
+            | 'PhoneNumber'
+            | 'ReportCategory'
+            | 'Address'
+            | 'GpsCoordinates'
+            | 'IssueDescription'
+            | 'Attachments'
+            | 'Email'
+            | 'Sms'
+            | 'StatusOfReport'
+            | 'DateTimeOfSubmission'
+        > = {
+            ReportID: {
+                S: reportId,
+            },
+        };
+
+        const getData = await this.ddbClient.getItem({
```

Comment:
Is there appropriate error handling anywhere if we're getting a report that does not exist?

Reply:
![image](https://github.com/CodeDayLabs311/cl311/assets/123401167/67a2cb5e-0a09-401f-b61b-ce497edcb85c)
I tested it. It worked.